src/core/Controls.ts

```diff
@@ -1,24 +1,75 @@
 import * as THREE from "three";
-
+import { Game } from "./Game";
 /**
  * An instance of this class perceives a user's actions within the game,
  * and controls the game input.
  */
 export class Controls {
-    private camera: THREE.PerspectiveCamera;
+    private static camera: THREE.PerspectiveCamera;
+    public static controls: any;
+    public static euler = new THREE.Euler( 0, 0, 0, 'YXZ' );
     public constructor() {
-        this.camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 1000);
+        Controls.camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 1000);
+        Controls.createControls()
     }
     public getCamera(): THREE.Camera {
-        return this.camera;
+        return Controls.camera;
     }
     /**
      * Update this controls within the {@link context} specified.
      * @param context a context object to update the controls with
      */
     public update(context: { viewport: { width: number, height: number } }): void {
         const { width, height } = context.viewport;
-        this.camera.aspect = width / height;
-        this.camera.updateProjectionMatrix();
+        Controls.camera.aspect = width / height;
+        Controls.camera.updateProjectionMatrix();
+    }
+    public static createControls(): void {
+        let speed = 10, sensitivity = 1
+        // flyControls for comfortable devloping   
+        let canvas = document.querySelector('#canvas') as HTMLCanvasElement;
+          const canvasClick = () => {
```

Comment:
create a respective method on this class, and pass its reference to the `<model>.addEventListener()`'s 2nd argument with help of `.bind(this)`, e.g.
```typescript
private _onDomElementClick(): void {
    this.domElement.requestPointerLock();
}
private _createControls(): void {
    this.domElement.addEventListener('click', this._onDomElementClick.bind(this), false);
}
```